src/apps/components/AppAlerts/utils.ts

```diff
@@ -0,0 +1,44 @@
+import { AppEventDeliveriesFragment, EventDeliveryStatusEnum } from "@dashboard/graphql";
+import moment from "moment";
+
+export type Webhook = NonNullable<AppEventDeliveriesFragment["webhooks"]>[0];
+
+const hasFailedAttemptsCheck = (webhook: Webhook) =>
+  webhook.failedDelivers && webhook.failedDelivers?.edges?.length > 0;
+const hasFailedAttemptsInPendingCheck = (webhook: Webhook) => {
+  const preliminaryCheck = webhook.pendingDelivers && webhook.pendingDelivers?.edges?.length > 0;
+
+  if (!preliminaryCheck) return false;
+
+  return webhook.pendingDelivers?.edges.some(edge =>
+    edge.node?.attempts?.edges.some(
+      attempt => attempt.node.status === EventDeliveryStatusEnum.FAILED,
+    ),
+  );
+};
+
+export const webhookFailedAttemptsCheck = (webhook: Webhook) =>
+  hasFailedAttemptsCheck(webhook) || hasFailedAttemptsInPendingCheck(webhook);
+
+const getLatestFailedAttemptFromWebhook = (webhook: Webhook) => {
+  const fromFailedDelivers = webhook.failedDelivers?.edges?.[0]?.node;
+  const fromPendingDelivers = webhook.pendingDelivers?.edges?.[0]?.node.attempts?.edges?.[0]?.node;
+
+  if (fromFailedDelivers && fromPendingDelivers) {
+    return moment(fromFailedDelivers?.createdAt).isAfter(moment(fromPendingDelivers?.createdAt))
+      ? fromFailedDelivers
+      : fromPendingDelivers;
+  } else if (fromFailedDelivers) {
+    return fromFailedDelivers;
+  } else if (fromPendingDelivers) {
+    return fromPendingDelivers;
+  } else {
+    return null;
+  }
```

Comment:
suggestion: this is unreadable
try something like this

```suggestion
 if (fromFailedDelivers && fromPendingDelivers) {
    return someCheck()
  } 
  
    if (fromFailedDelivers) {
    return fromFailedDelivers;
  } 
  
    if (fromPendingDelivers) {
    return fromPendingDelivers;
  } 
 
   return null;
```